### PR TITLE
Fix README rand_sparse_tri QuickStart example (#92) 

### PR DESCRIPTION
**Pull request by NusaybahA**
Summary

- Replaced the invalid `unit_triangular` parameter with `strict` in the README Quickstart examples to align with the current `rand_sparse_tri()` API.
- Updated example usage to reflect the correct supported parameters.
- Clarified the sparsity comment to accurately describe the generated 1000×1000 dense matrix size and avoid confusion around `event_size=1000`.

Testing

- Verified that the README examples now use the correct `strict` keyword.
- Confirmed the sparsity comment accurately reflects the matrix dimensions and sparsity ratio.

### CODE REVIEW

**Comment by aymuos15** [comment]
Congrats on your first contribution! 🎉

The `strict` fix is correct, and `requires_grad_(True)` fixes a second latent bug: without it, the trailing `scale_tril.grad.shape` line hits `AttributeError` on `None`. I ran the snippet end-to-end and it works now.

Maybe another small change - Three comments still say "unit triangular," but `strict=True` gives a zero diagonal, not a unit one.

```diff
-# Create sparse lower triangular matrix (unit triangular, no diagonal)
+# Create sparse lower triangular matrix (strictly lower triangular, no diagonal)
```

```diff
 dist_ldlt = SparseMultivariateNormal(
     loc=loc,
     diagonal=diagonal,
-    scale_tril=scale_tril  # Unit lower triangular
+    scale_tril=scale_tril  # Strictly lower triangular
 )
```

```diff
 dist_precision = SparseMultivariateNormal(
     loc=loc,
     diagonal=precision_diagonal,
-    precision_tril=precision_tril  # Unit triangular precision factor
+    precision_tril=precision_tril  # Strictly triangular precision factor
 )
```

**Comment by NusaybahA** [comment]
After reviewing the code I updated any remaining comments that included "unit triangular" ensuring they align with the rest of the code and so that the comments are clear to read.

**Comment by aymuos15** [comment]
@theo-barfoot LGTM

**Review by theo-barfoot — CHANGES_REQUESTED** [review]
Thanks for this, this looks good. The replacement of `unit_triangular` with `strict` correctly matches the current `rand_sparse_tri` API, and the values used for the LDLᵀ and LLᵀ examples are appropriate. Adding `requires_grad_(True)` also fixes the gradient section at the end of the example.

I found one small pre-existing terminology issue, that was my mistake, which it would be good to correct. In the updated sparsity comment: 5,000 nonzeros in a 1000 × 1000 matrix corresponds to 0.5% density, or 99.5% sparsity. 

So I would change the comment to: 

`nnz=5000, # 5000 nonzeros per 1000x1000 matrix (99.5% sparsity)`

Once that wording is corrected, this looks ready to merge.

**Comment by NusaybahA** [comment]
I have now corrected the sparsity comment to state a 99.5% sparsity, to make the wording clear.

**Comment by aymuos15** [comment]
Thanks for the review @theo-barfoot !

**Review by theo-barfoot — APPROVED** [review]
Looks good! Congrats on your first contribution 😃
